fix(FocusManager): Replaces implementation with focus-trap-react

JoaoTMDias opened the pull request:
## Because
In order to fix the bug reported on issue 2

## This Commit
Does a re-implementation of the `FocusManager`, by replacing the current functionality, with the functionality provided by the ["focus-trap-react" package](https://www.npmjs.com/package/focus-trap-react).

In order to keep the same API as before, it makes them still available as before, but changes the default values for `autoFocus`, `restoreFocus` and `contain` props. It also exposes the "focus-trap" configuration API on the `options` prop.

As such, the component itself also gains a whole set of new functionalities, made possible by the `focus-trap` package.

BREAKING CHANGE: autoFocus, restoreFocus and contain are now set to true by default
BREAKING CHANGE: the `useFocusManager` was reimplemented. Since we no longer use a React context state management solution to travel between elements, the hook was re-implemented as an optional way to facilitate the creation of a `FocusTrap` instance, but without using the provided element by the package.

Closes #27

tiagocastro070 commented on src/components/focus-manager/helpers/index.ts:
```diff
@@ -0,0 +1,53 @@
+	autoFocus: FocusManagerProps["autoFocus"],
+	initialFocus: Options["initialFocus"],
+) {
+	if (autoFocus === true) {
```
Isn't this enough?
```js
if (autoFocus) {
```

JoaoTMDias replied:
I've changed the implementation of this to be more clear:
```ts
/**
 * Sets the initial focus element.
 */
export function getInitialFocus(
	autoFocus: FocusManagerProps["autoFocus"],
	initialFocus: Options["initialFocus"],
) {
	// If true, resumes the normal behaviour
	if (autoFocus === true) {
		return undefined;
	}

	// If false, prevents the autoFocus from happening
	else if (autoFocus === false) {
		return false;
	}

	// If is nil, uses the value defined in the `initialFocus` param,
	// which if also nil, resumes the normal behaviour
	return initialFocus;
}
```

tiagocastro070 commented on src/components/focus-manager/useFocusManager.tsx:
```diff
@@ -4,52 +4,91 @@
+	props: Omit<FocusManagerProps, "children">,
+	ref?: never,
+): UseFocusManagerAsProps;
+export function useFocusManager<GenericElement extends HTMLElement | SVGElement>(
```
Was this a mistake? Exporting 3 functions with the name `useFocusManager` on the same file.

JoaoTMDias replied:
Nope, it is a typescript feature [called function overloading](https://blog.logrocket.com/implementing-function-overloading-typescript/).
Basically, you can define multiple function signatures so that, depending on the input, the expected output is different. 
For example, if you pass only the props, the hook behaves as if inside a component, otherwise it will behave like a hook that provides the `FocusTrap` functionality.

Commit:
fix(FocusManager): Replaces implementation with focus-trap-react

In order to fix the bug reported on issue 2, this commit does a re-implementation of the FocusManager, by replacing the current
functionality, with the functionality provided by the "focus-trap" package.

In order to keep the same API as before, it makes them still available as before, but changes the default values for autoFocus, restoreFocus and contain props.

It also exposes the "focus-trap" API on the "options".

BREAKING CHANGE: autoFocus, restoreFocus and contain are now set to true by default

Closes #27

CarolinaRMarques commented:
I can’t approve here, but from my side, everything seems good to go! 

One suggestion for improvement would be to include a small video or screenshot of the demo. While this may not be necessary for this particular case, I believe it can always help people understand the context a bit better. This is just a personal preference, but I think it would be beneficial for future PRs.

Nonetheless, great job!!
